server/subscription_migration.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/mattermost/mattermost-plugin-confluence/server/serializer"
+)
+
+const (
+	createSubscriptionPath    = "/api/v1/instance/%s/%s"
+	subscriptionCreatedHeader = "\nThe list of subscriptions created successfully:-\n"
+	subscriptionFailedHeader  = "\nThe list of failed subscriptions:-\n"
+)
+
+type MigrateSubscriptionBody struct {
+	SubscriptionType string   `json:"subscriptionType"`
+	Alias            string   `json:"alias"`
+	BaseURL          string   `json:"baseURL"`
+	SpaceKey         string   `json:"spaceKey"`
+	PageID           string   `json:"pageID"`
+	ChannelID        string   `json:"channelID"`
+	Events           []string `json:"events"`
+}
+
+func (p *Plugin) migrateSubscription(subscriptions []serializer.Subscription, userID string) string {
+	subscriptionCreated := subscriptionCreatedHeader
+	failedSubscription := subscriptionFailedHeader
+	for _, sub := range subscriptions {
+		failedSubscriptionLog := ""
+		requestPayload, err := json.Marshal(&MigrateSubscriptionBody{
+			SubscriptionType: sub.Name(),
+			Alias:            sub.GetAlias(),
+			BaseURL:          sub.GetConfluenceURL(),
+			SpaceKey:         sub.GetSpaceKeyOrPageID(),
+			PageID:           sub.GetSpaceKeyOrPageID(),
+			ChannelID:        sub.GetChannelID(),
+			Events:           sub.GetEvents(),
+		})
+		if err != nil {
+			failedSubscriptionLog = err.Error()
+			p.API.LogError("Unable to marshal request body for subscription", "Subscription", sub, "Error", err.Error())
```

Comment:
Shouldn't there be a `continue` statement after this line?
You should append the error in `failedSubscription` variable and then `continue` after this line.